content/blog/2023/07/22/2023-07-22-gsoc-2023-midterm.adoc

```diff
@@ -0,0 +1,183 @@
+
+Sharing details about the probes that I worked on during the first phase of GSoC.
+
+=== Unreleased Production Changes Probe
+
+A bug fix, security fix, or new features are of no use unless they are released.
+The "unreleased production changes" probe looks for unreleased commits in the production files.
+
+The probe aims to identify unreleased production commits in the Maven production files.
+
+My mentors suggested I should start with writing the test cases to reproduce the probe requirements, and later add code to pass each of the test cases.
+
+I implemented the JGit library and understood its features.
+I lacked depth in Git concepts that made me struggle with JGit.
+This led me to strengthen the Git concepts along the way.
+
+=== The Third-Party Repository Detection Probe
+
+By definition, a third-party repository is a repository that is not hosted at link:https://repo.jenkins-ci.org/[].
+
+The third-party repository detection probe ensures that plugins are built only from trusted and reliable Maven repositories.
+
+This probe was requested by the community.
+
+A major challenge in implementing this probe was considering the parent and child pom relationship from all angles.
+
+Some edge-case scenarios were not considered while specifying the probe.
+And, due to the number of open questions, this probe was put on hold until further research was completed.
+
+In the process, I have extensively read the Maven documentation to understand how to test POM structure, inheritance, and Maven API.
+
+=== GitHub Security Scan Workflow Probe
+
+The Security Scan probe identifies whether the plugin is configured to execute the security action.
+
+This probe is important from the security perspective, as it looks for known security breaches and vulnerabilities in a plugin.
+
+Its primary purpose is to verify that the security scan is properly configured in the plugin's GitHub Action.
+
+This probe is also involved in the refactoring of existing classes to avoid code duplication.
+The JEP-229 (Continuous  Delivery) probe looks for CI configurations in GitHub Actions as well.
+
+Thus, I have gained knowledge in leveraging the class hierarchy to build an abstract probe that can be implemented by any probes requiring GitHub Action configuration checks.
+
+This probe will be beneficial to the security team as it helps to identify any gaps and helps to improve the security of the Jenkins plugin ecosystem.
+
+=== What about the second-half of GSoC?
+
+In the second half of the GSoC I will work on a few more probes like the following ones:
+
+* Detect renovate usage
+** Detects how many plugins use the renovate bot CLI to detect outdated dependencies in the plugin.
+** If it finds any, it automatically creates a pull request (PR) in which the dependency is updated.
+* Count open tickets
+** Detect open issues in Github and JIRA
+
+Also will work on fixing bugs and making enhancements to the existing probes.
```

Comment:
```suggestion
In the second half of the GSoC project, I will be working on a few more probes, including the following:

* Detect Renovate usage:
** This probe will identify how many plugins use the Renovate bot CLI to detect outdated dependencies in the plugin.
** If any outdated dependencies are found, it will automatically create a pull request (PR) to update the dependency.
* Count open tickets:
** This probe will detect open issues in both Github and JIRA.

In addition to these new probes, I will also be focusing on fixing bugs and making enhancements to the existing probes.
```